internal/cmdrender/executor.go

```diff
@@ -54,12 +57,21 @@ func (e *Executor) Execute(ctx context.Context) error {
 
 	// initialize hydration context
 	hctx := &hydrationContext{
-		root: root,
-		pkgs: map[types.UniquePath]*pkgNode{},
+		root:      root,
+		pkgs:      map[types.UniquePath]*pkgNode{},
+		fnResults: fnresult.NewResultList(),
 	}
 
 	resources, err := hydrate(ctx, root, hctx)
 	if err != nil {
+		resultsFile, resultsErr := fnruntime.SaveResults(e.ResultsDirPath, hctx.fnResults)
+		if resultsErr != nil {
+			return fmt.Errorf("failed to save function results: %w", err)
+		}
+
+		if resultsFile != "" {
+			pr.Printf("Results saved successfully at %q.\n", resultsFile)
```

Comment:
nit, slightly better wording

`For complete results, see /some/path/function-results.yaml`

Reply:
Done.